solr/core/src/test/org/apache/solr/TestRandomDVFaceting.java

```diff
@@ -125,7 +125,7 @@ void deleteSomeDocs() {
     int percent = rand.nextInt(100);
     if (model == null) return;
     ArrayList<String> ids = new ArrayList<>(model.size());
-    for (@SuppressWarnings({"rawtypes"})Comparable id : model.keySet()) {
+    for (Comparable<?> id : model.keySet()) {
```

Comment:
observation: `model` itself retains its rawtypes suppress warnings for now.

Reply:
Yes, all the Comparable stuff in Facets and Groups and Fields is going to need to be it's own issue. I'm not entirely certain how to deal with it at this point, and some of it depends on what Lucene does with their APIs